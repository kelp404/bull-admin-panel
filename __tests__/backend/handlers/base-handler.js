const baseHandler = require('../../../lib/handlers/base-handler');
const {flushDbThenGenerateQueues} = require('../utils');
let queues;

beforeEach(async () => {
  queues = await flushDbThenGenerateQueues();
});

afterEach(() => {
  jest.restoreAllMocks();
  queues.forEach(queue => queue.close());
});

describe('get base template', () => {
  test('by defaults', () => {
    const req = {};
    const res = {
      locals: {
        bullAdminPanel: {
          assetsPath: '/assets',
          basePath: '/bull',
        },
      },
      send: jest.fn(html => expect(html).toMatchSnapshot()),
    };

    baseHandler.getBaseTemplate(req, res);
    expect(res.send).toBeCalled();
  });
});
