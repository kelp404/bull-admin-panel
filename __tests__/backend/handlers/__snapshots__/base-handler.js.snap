// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get base template by defaults 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\"/>
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no\\"/>
  <title>Bull Admin Panel</title>

  <link type=\\"text/css\\" rel=\\"stylesheet\\" href=\\"/assets/web.css\\"/>
  <script type=\\"text/javascript\\">
    function base64DecodeUnicode(text) {
      return decodeURIComponent(atob(text).split('').map(function (c) {
        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
      }).join(''));
    }
    var config = JSON.parse(base64DecodeUnicode('eyJhc3NldHNQYXRoIjoiL2Fzc2V0cyIsImJhc2VQYXRoIjoiL2J1bGwifQ=='));
  </script>
</head>
<body>
<div id=\\"root\\">
  <div class=\\"text-center text-muted py-5 main-content\\">
    <div class=\\"spinner-border\\">
      <span class=\\"sr-only\\">Loading...</span>
    </div>
  </div>
</div>
<script type=\\"text/javascript\\" src=\\"/assets/web.js\\"></script>
</body>
</html>
"
`;
